keel-sql/src/main/kotlin/com/netflix/spinnaker/keel/sql/SqlStorageContext.kt

```diff
@@ -0,0 +1,20 @@
+package com.netflix.spinnaker.keel.sql
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.netflix.spinnaker.keel.api.plugins.ArtifactSupplier
+import com.netflix.spinnaker.keel.resources.ResourceSpecIdentifier
+import com.netflix.spinnaker.keel.resources.SpecMigrator
+import org.jooq.DSLContext
+import java.time.Clock
+
+abstract class SqlStorageContext(
+  internal val jooq: DSLContext,
+  internal val clock: Clock,
+  internal val sqlRetry: SqlRetry,
+  internal val objectMapper: ObjectMapper,
+  internal val resourceSpecIdentifier: ResourceSpecIdentifier,
+  internal val artifactSuppliers: List<ArtifactSupplier<*, *>>,
+  internal val specMigrators: List<SpecMigrator<*, *>>
+) {
+  internal val resourceFactory = ResourceFactory(objectMapper, resourceSpecIdentifier, specMigrators)
+}
```

Comment:
This is the start of some refactoring I've been talking about. For now I have some of the repositories extend this, which allows them to share some JOOQ / SQL code that is useful in multiple repository contexts. Eventually I'd like to get rid of the repositories altogether and replace them with a library of extension methods on this.